README.md

```diff
@@ -58,6 +61,39 @@ func main() {
 }
 ```
 
+### Maintaining a Channel Subscription
+By default, the client will *reconnect but not resubscribe to your previously-subscribed channels* if a network interruption or other disconnection occurs.  This is the intended behavior and is aligned with the other Emitter and generic MQTT clients.  There are two ways to maintain your channel subscriptions across reconnections:
+
+#### Method 1: Subscribe in the OnConnect() handler:
+```go
+// Subscribe to the channels in OnConnect() so they are re-subscribed on reconnect
+c.OnConnect(func(c *emitter.Client) {
+	c.Subscribe(key, channel, nil)
+})
+
+// Create the client and connect to the broker
+c, _ := emitter.Connect("", func(_ *emitter.Client, msg emitter.Message) {
+    fmt.Printf("Received: '%s' topic: '%s'\n", msg.Payload(), msg.Topic())
+})
+```
+
+#### Method 2: Use Persistent Connections:
```

Comment:
I don't think we should be mentioning this here, given that Emitter doesn't support it (and this is an emitter-specific library).

Reply:
Could you elaborate? Are you saying the documentation shouldn't give users a hint on how to achieve persistent connections? I think the docs ought to be as helpful as reasonably possible. Apologies if I misunderstood you.

Reply:
The `cleanSession` flag work on traditional MQTT brokers, but it's not implemented in Emitter broker. So if you mention persistent connections in here, people would think that it actually works with Emitter.

Reply:
Oh, thanks for the clarification @kelindar, I didn't realize that it doesn't work in Emitter.  Perhaps there is no point in adding `WithCleanSession()` either then?  I definitely think the first method (subscribing in `OnConnect()`) is important, otherwise there is no way to be sure you are even subscribed a second later since it silently reconnects without resubscribing.  This problem is at the root of the PR since I have found that running dozens of clients throughout the world, with an AWS ALB in front of emitter caused clients to disconnect occasionally, until eventually there were no clients left at all.

Reply:
Well, that's what we discussed from the start. It seems you wanted to add this `WithCleanSession` anyway, because the go library actually resubscribe - according to the sources you cited. But, as I said earlier, it's specific to the go library, and the usual pattern should be to subscribe in the `OnConnect`